src/processor/operator/physical_operator.cpp

```diff
@@ -185,6 +185,19 @@ bool PhysicalOperator::getNextTuple(ExecutionContext* context) {
     return result;
 }
 
+void PhysicalOperator::finalize(ExecutionContext* context) {
+    if (hasBeenFinalized) {
+        return;
+    }
+    hasBeenFinalized = true;
```

Comment:
The addition of the recursive calls to finalize can cause finalizeInternal() can currently be called multiple times (across pipelines). This hack prevents any issues that this can cause, but we should properly handle it by ensuring finalize() is called at most once per operator